src/app/Plans/components/helpers.ts

```diff
@@ -1,92 +1,175 @@
-  | 'Copying'
-  | 'AbortedCopying'
-  | 'StartingCutover'
-  | 'Cutover'
-  | 'Finished';
-
-export const getWarmPlanState = (
+export const getMigStatusState = (state: PlanState | null, isWarmPlan: boolean) => {
+  let title = '';
+  let variant: ProgressVariant | undefined;
+
+  switch (true) {
+    case state === 'Starting': {
+      title = 'Running';
+      break;
+    }
+    case state === 'NotStarted': {
+      title = '';
+      break;
+    }
+    case state === 'Finished-Failed':
+    case state === 'AbortedCopying': {
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      title = 'Succeeded';
+      variant = ProgressVariant.success;
+      break;
+    }
+    case state === 'Copying':
+    case state === 'PipelineRunning': {
+      title = isWarmPlan ? 'Running cutover' : 'Running';
+      break;
+    }
+    case state === 'Finished-Incomplete': {
+      title = 'Finished - Incomplete';
+      variant = ProgressVariant.warning;
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
+
+export const getButtonState = (state: PlanState | null): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case state === 'Starting': {
+      type = null;
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'Finished-Incomplete':
+    case state === 'Finished-Failed':
+    case state === 'Canceled':
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null;
+      break;
+    }
+    case state === 'PipelineRunning': {
+      type = null;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      type = null;
+      break;
+    }
+    default: {
+      type = null;
+    }
+  }
+
+  return type;
+};
+
+export const getPlanState = (
   plan: IPlan | null,
   migration: IMigration | null,
-  migrationsQuery: UseQueryResult<IKubeList<IMigration>>
-): WarmPlanState | null => {
-  if (!plan || !plan.spec.warm) return null;
-  if (!migration) return 'NotStarted';
-  if (isPlanBeingStarted(plan, migration, migrationsQuery)) {
-    return 'Starting';
-  }
+  migrationQuery: UseQueryResult<IKubeList<IMigration>>
+): PlanState | null => {
+  if (!plan) return null;
+  const isWarm = plan.spec.warm;
+
+  if (!migration || !plan.status?.migration?.started) return 'NotStarted';
+  if (isPlanBeingStarted(plan, migration, migrationQuery)) return 'Starting';
+
   const conditions = plan.status?.conditions || [];
-  if (
-    (hasCondition(conditions, PlanStatusType.Canceled) ||
-      hasCondition(conditions, PlanStatusType.Failed)) &&
-    !migration.spec.cutover
-  ) {
-    return 'AbortedCopying';
+
+  if (isWarm) {
```

Comment:
If you're not going to combine this case with the other isWarm cases, you can just put `isWarm &&` into the if statement you have inside here instead of wrapping it